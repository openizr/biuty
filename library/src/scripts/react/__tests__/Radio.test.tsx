/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import UIRadio from 'scripts/react/Radio';
import { act, Simulate } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');
jest.mock('scripts/helpers/generateRandomId');
const options: { value: string; label: string; disabled?: boolean; }[] = [
  { value: 'option1', label: 'Option 1' },
  { value: 'option2', label: 'Option 2' },
  { value: 'option3', label: 'Option 3' },
  { value: 'option4', label: 'Option 4' },
];

describe('react/UIRadio', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<UIRadio name="test" modifiers="large" options={options} />, container);
    });
    const input = document.querySelector('input') as HTMLInputElement;
    act(() => {
      Simulate.focus(input);
    });
    act(() => {
      Simulate.change(input);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    act(() => {
      render(<UIRadio name="test" id="test" options={options} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with label', () => {
    act(() => {
      render(<UIRadio name="test" label="test" options={options} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with helper', () => {
    act(() => {
      render(<UIRadio name="test" helper="test" options={options} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with value', () => {
    act(() => {
      render(<UIRadio name="test" value="option1" options={options} />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - disabled', () => {
    act(() => {
      render(<UIRadio name="test" options={options} modifiers="disabled" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with option disabled', () => {
    act(() => {
      render(
        <UIRadio
          name="test"
          options={[{ value: 'option5', label: 'Option 5', disabled: true }]}
        />,
        container,
      );
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with listeners', () => {
    const onChange = jest.fn();
    const onFocus = jest.fn();
    act(() => {
      render(
        <UIRadio
          name="test"
          value="option2"
          options={options}
          onChange={onChange}
          onFocus={onFocus}
        />,
        container,
      );
    });
    const input = document.querySelector('input') as HTMLInputElement;
    act(() => {
      Simulate.focus(input);
    });
    act(() => {
      Simulate.change(input);
    });
    expect(container).toMatchSnapshot();
    expect(onFocus).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledWith('option1');
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('option1');
  });
});
