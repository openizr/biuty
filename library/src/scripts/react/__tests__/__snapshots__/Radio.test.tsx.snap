// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/UIRadio renders correctly - basic 1`] = `
<div>
  <div
    class="ui-radio ui-radio--large"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option ui-radio__wrapper__option--checked"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - disabled 1`] = `
<div>
  <div
    class="ui-radio ui-radio--disabled"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="-1"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="-1"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="-1"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="-1"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with helper 1`] = `
<div>
  <div
    class="ui-radio"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
    <span
      class="ui-radio__wrapper__helper"
    >
      test
    </span>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with id 1`] = `
<div>
  <div
    class="ui-radio"
    id="test"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with label 1`] = `
<div>
  <div
    class="ui-radio"
  >
    <label
      class="ui-radio__label"
      for="z8a8d7a5zad4"
    >
      test
    </label>
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with listeners 1`] = `
<div>
  <div
    class="ui-radio"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option ui-radio__wrapper__option--checked"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          checked=""
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with option disabled 1`] = `
<div>
  <div
    class="ui-radio"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option ui-radio__wrapper__option--disabled"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          disabled=""
          name="test"
          tabindex="-1"
          type="radio"
          value="option5"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 5
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`react/UIRadio renders correctly - with value 1`] = `
<div>
  <div
    class="ui-radio"
  >
    <div
      class="ui-radio__wrapper"
    >
      <label
        class="ui-radio__wrapper__option ui-radio__wrapper__option--checked"
      >
        <input
          checked=""
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option1"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 1
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option2"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 2
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option3"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 3
        </span>
      </label>
      <label
        class="ui-radio__wrapper__option"
      >
        <input
          class="ui-radio__wrapper__option__radio"
          name="test"
          tabindex="0"
          type="radio"
          value="option4"
        />
        <span
          class="ui-radio__wrapper__option__label"
        >
          Option 4
        </span>
      </label>
    </div>
  </div>
</div>
`;
